addons/sale_stock/tests/test_sale_stock.py

```diff
@@ -1143,6 +1143,80 @@ def test_exception_delivery_partial_multi(self):
+            'group_stock_tracking_lot': True,
+            'group_stock_adv_location': True,
+            'group_stock_multi_locations': True,
+        })
+        warehouse.write({
+            'delivery_steps': 'pick_pack_ship',
+        })
+        self.env['stock.quant']._update_available_quantity(self.test_product_delivery, warehouse.lot_stock_id, 10)
+
+        so_1 = self._get_new_sale_order(product=self.test_product_delivery)
+        so_1.action_confirm()
+        pick_picking = so_1.picking_ids.filtered(lambda x: x.picking_type_id == warehouse.pick_type_id)
+        pack_picking = so_1.picking_ids.filtered(lambda x: x.picking_type_id == warehouse.pack_type_id)
+        out_picking = so_1.picking_ids.filtered(lambda x: x.picking_type_id == warehouse.out_type_id)
+
+        pick_picking.move_lines.write({'quantity_done': 2})
+        pick_picking.action_put_in_pack()
+        backorder_wizard_dict = pick_picking.button_validate()
+        backorder_wizard = Form(
+            self.env[backorder_wizard_dict['res_model']].with_context(backorder_wizard_dict['context'])).save()
+        backorder_wizard.process()
+
+        pack_picking.move_line_ids.write({'result_package_id': False})
+        pack_picking.move_lines.write({'quantity_done': 2})
+        backorder_wizard_dict = pack_picking.button_validate()
+        backorder_wizard = Form(
+            self.env[backorder_wizard_dict['res_model']].with_context(backorder_wizard_dict['context'])).save()
+        backorder_wizard.process()
+
+        self.assertEqual(out_picking.move_line_ids.package_id.id, False)
+        self.assertEqual(out_picking.move_line_ids.result_package_id.id, False)
+
+        pick_picking_2 = so_1.picking_ids.filtered(
+            lambda x: x.picking_type_id == warehouse.pick_type_id and x.state != 'done')
+
+        pick_picking_2.move_lines.write({'quantity_done': 2})
+        package_2 = pick_picking_2.action_put_in_pack()
+        backorder_wizard_dict = pick_picking_2.button_validate()
+        backorder_wizard = Form(
+            self.env[backorder_wizard_dict['res_model']].with_context(backorder_wizard_dict['context'])).save()
+        backorder_wizard.process()
+
+        self.assertEqual(out_picking.move_line_ids.package_id.id, False)
+        self.assertEqual(out_picking.move_line_ids.result_package_id.id, False)
+
+        pack_picking_2 = so_1.picking_ids.filtered(
+            lambda x: x.picking_type_id == warehouse.pack_type_id and x.state != 'done')
+
+        pack_picking_2.move_lines.write({'quantity_done': 2})
+        backorder_wizard_dict = pack_picking_2.button_validate()
+        backorder_wizard = Form(
+            self.env[backorder_wizard_dict['res_model']].with_context(backorder_wizard_dict['context'])).save()
+        backorder_wizard.process()
+
+        self.assertEqual(len(out_picking.move_line_ids), 2)
+        self.assertEqual(len(out_picking.move_line_ids.filtered(lambda x: x.result_package_id == package_2)), 1)
```

Comment:
You can eventually use `assertRecordValues` who will check how many records you have and the fields' value of these records
```suggestion
        self.assertRecordValues(out_picking.move_line_ids, [
            {'result_package_id': False}, {'result_package_id': package_2.id}])
```
It's maybe a little more precise than checking the length of the filtered records, but honestly it doesn't change so much here so do like you prefer :shrug: 

It looks good to me otherwise :)